subsys/bluetooth/host/direction.c

```diff
@@ -270,10 +312,53 @@ int bt_df_set_adv_cte_tx_param(struct bt_le_ext_adv *adv,
 		return -EINVAL;
 	}
 
+	if (atomic_test_bit(adv->flags, BT_PER_ADV_CTE_ENABLED)) {
+		return -EINVAL;
+	}
+
 	err = hci_df_set_cl_cte_tx_params(adv, params);
 	if (err) {
 		return err;
 	}
 
+	atomic_set_bit(adv->flags, BT_PER_ADV_CTE_PARAMS_SET);
+
+	return 0;
+}
+
+static int bt_le_df_set_adv_cte_tx_enabled(struct bt_le_ext_adv *adv,
+					   bool enable)
+{
+	int err;
+
+	if (!atomic_test_bit(adv->flags, BT_PER_ADV_PARAMS_SET)) {
+		return -EINVAL;
+	}
+
+	if (!atomic_test_bit(adv->flags, BT_PER_ADV_CTE_PARAMS_SET)) {
+		return -EINVAL;
+	}
+
+	if (enable == atomic_test_bit(adv->flags, BT_PER_ADV_CTE_ENABLED)) {
+		return -EALREADY;
+	}
+
+	err = hci_df_set_adv_cte_tx_enable(adv, enable);
+	if (err) {
+		return err;
+	}
+
 	return 0;
```

Comment:
Same here, since you're not really doing anything with `err`

Reply:
Corrected according to suggestion